File: frontend/src/types/Safety.ts
export type RiskLevel = "low" | "moderate" | "high";
export type EvidenceGrade = "A" | "B" | "C" | "D";
export type BoundaryCategory = "informational" | "lifestyle" | "experiment";

export interface SafetyIssue {
  code: string;
  severity: RiskLevel;
  message: string;
  details?: Record<string, any>;
}

export interface SafetyDecision {
  allowed: boolean;
  risk: RiskLevel;
  boundary: BoundaryCategory;
  evidence_grade: EvidenceGrade;
  issues: SafetyIssue[];
}

export interface SafetyEvaluateRequest {
  intervention_key: string;
  user_flags?: Record<string, any>;
  requested_boundary?: string;
  requested_evidence?: string;
}

File: backend/app/api/schemas/safety.py
from __future__ import annotations

from typing import Any, Dict, List, Literal, Optional

from pydantic import BaseModel, Field


class SafetyIssueOut(BaseModel):
    code: str
    severity: Literal["low", "moderate", "high"]
    message: str
    details: Optional[Dict[str, Any]] = None


class SafetyDecisionOut(BaseModel):
    allowed: bool
    risk: Literal["low", "moderate", "high"]
    boundary: Literal["informational", "lifestyle", "experiment"]
    evidence_grade: Literal["A", "B", "C", "D"]
    issues: List[SafetyIssueOut] = Field(default_factory=list)


class SafetyEvaluateRequest(BaseModel):
    intervention_key: str
    user_flags: Optional[Dict[str, Any]] = None
    requested_boundary: Optional[str] = None
    requested_evidence: Optional[str] = None

